shim/src/block-imports.js

```diff
@@ -0,0 +1,19 @@
+// this \s *must* match all kinds of syntax-defined whitespace. If e.g.
+// U+2028 (LINE SEPARATOR) or U+2029 (PARAGRAPH SEPARATOR) is treated as
+// whitespace by the parser, but not matched by /\s/, then this would admit
+// an attack like: import\u2028('power.js') . We're trying to distinguish
+// something like that from something like importnotreally('power.js') which
+// is perfectly safe.
+
+const scanner = /^(.*)\bimport\s*(\(|\/\/|\/\*)/m;
```

Comment:
Why not use the regex constructor? It makes code more readable.

Reply:
I'd be happy to use that (I assume it has the same concerns about poisoned primordials as using `//`), what would the syntax be?